monkey/monkey_island/cc/server_utils/encryption/repository_encryptor.py

```diff
@@ -1,44 +1,44 @@
 import secrets
 from pathlib import Path
+from typing import Optional
 
 from monkey_island.cc.server_utils.encryption.encryption_key_types import EncryptionKey32Bytes
 from monkey_island.cc.server_utils.file_utils import open_new_securely_permissioned_file
 
-from . import ILockableEncryptor, LockedKeyError, ResetKeyError, UnlockError
+from . import IEncryptor, ILockableEncryptor, LockedKeyError, ResetKeyError, UnlockError
 from .key_based_encryptor import KeyBasedEncryptor
 from .password_based_bytes_encryptor import PasswordBasedBytesEncryptor
 
 
 class RepositoryEncryptor(ILockableEncryptor):
     def __init__(self, key_file: Path):
         self._key_file = key_file
-        self._password_based_encryptor = None
-        self._key_based_encryptor = None
+        self._key_based_encryptor: Optional[IEncryptor] = None
 
     def unlock(self, secret: bytes):
         try:
-            self._password_based_encryptor = PasswordBasedBytesEncryptor(secret.decode())
-            self._key_based_encryptor = self._initialize_key_based_encryptor()
+            encryptor = PasswordBasedBytesEncryptor(secret.decode())
+            self._key_based_encryptor = self._initialize_key_based_encryptor(encryptor)
```

Comment:
Why is this change needed?

Reply:
This is being replaced anyway.

Reply:
What prompted the change was that, due to temporal coupling, `self._password_based_encryptor` could be `None`, so we couldn't call `decrypt()` or `encrypt()`. It turns out that we don't need to store the `PasswordBasedBytesEncryptor` at all since we only use it here. That way, it's never `None`. Also, if we don't have a reason to hold onto it, then we shouldn't hold onto it